### DISCUSSION TITLE
Integrate TLO measurement into core

### DISCUSSION THREAD

**Discussion by andrewmarles** [grblHAL/core#129] [Ideas]
Right now, with IOSender I am using the probing routines to reference the TLO at the G59.1 position.  This works very well and and the semi-auto toolchange integration in grblhal is so smooth.  Since the TLO probe is essentially the same probing function as the semi-auto tool change, would it be worth integrating this TLO reference measurement into the core and assigning it to an M code or something?  I think it should be able to use the same set of offsets and probing parameters as the toolchange and I think this would make semi-auto mode sender independent (M code could be added to post-processor) without having to use the MDI.

Just wondering if this sounds like a good idea or if I am missing something that necessitates the existing implementation. I am certainly willing to work on it.

**Comment by terjeio** [comment]
I have not been to keen to add "non-standard" M-codes to the core, for now I have implemented them in plugins.
Try to create a plugin first and then decide later if the code should be moved to the core?
